lib/core/include/irods/rodsError.h

```diff
@@ -33,7 +33,10 @@ typedef struct ErrorStack {
     struct ErrorMessage **errMsg;
 } rError_t;
 
-void allocateRErrorMsgIfNecessary(struct ErrorStack** myError);
+/// \brief Allocate memory for the ErrorStack struct if it is uninitialized
+///
+/// \param[in/out] myError the ErrorStack to be initialized
```

Comment:
It's difficult to express `T**` in words.

Consider including an example demonstrating how to use this function.

Perhaps something like:
```cpp
struct ErrorStack* p = NULL;
allocate_error_stack_if_necessary(&p);
assert(NULL != p);
```

Here's a link to how code examples can be included in documentation.
https://github.com/irods/irods/blob/8797c1c944175fe28024474b53cdef826672e74b/lib/core/include/irods/irods_logger.hpp#L565-L572